src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Diagnostics/DocumentPullDiagnosticsEndpointTest.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.LanguageServer.Diagnostics;
+using Microsoft.AspNetCore.Razor.Test.Common.LanguageServer;
+using Microsoft.AspNetCore.Razor.Test.Common.Workspaces;
+using Microsoft.CodeAnalysis.Razor.Workspaces;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Moq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Test.Diagnostics;
+
+public sealed class DocumentPullDiagnosticsEndpointTest(ITestOutputHelper testOutput) : LanguageServerTestBase(testOutput)
+{
+    [Fact]
+    public void ApplyCapabilities_AddsExpectedCapabilities()
+    {
+        // Arrange
+        var documentMappingService = new RazorDocumentMappingService(new FilePathService(TestLanguageServerFeatureOptions.Instance), new TestDocumentContextFactory(), LoggerFactory);
+        var featureOptions = new Mock<LanguageServerFeatureOptions>(MockBehavior.Strict);
+        var razorTranslate = new Mock<RazorTranslateDiagnosticsService>(MockBehavior.Strict,
+            documentMappingService,
+            LoggerFactory);
+        var clientConnection = new Mock<IClientConnection>(MockBehavior.Strict);
+        var endpoint = new DocumentPullDiagnosticsEndpoint(
+            featureOptions.Object,
+            razorTranslate.Object,
+            clientConnection.Object,
+            telemetryReporter: null);
```

Comment:
The `featureOptions` mock doesn't seem to be needed. You should be able to just pass `TestLanguageServerFeatureOptions.Instance`.

```suggestion
        var razorTranslate = new Mock<RazorTranslateDiagnosticsService>(MockBehavior.Strict,
            documentMappingService,
            LoggerFactory);
        var clientConnection = new Mock<IClientConnection>(MockBehavior.Strict);
        var endpoint = new DocumentPullDiagnosticsEndpoint(
            TestLanguageServerFeatureOptions.Instance,
            razorTranslate.Object,
            clientConnection.Object,
            telemetryReporter: null);
```

Reply:
Did you push the change?

Reply:
Unresolving because it appears that this change hasn't been pushed yet.

Reply:
There we go. :smile: